lms/services/vitalsource/model.py

```diff
@@ -0,0 +1,31 @@
+import re
+from dataclasses import dataclass
+
+
+@dataclass
+class VSBookLocation:
+    """A location within a VitalSource Book."""
+
+    book_id: str
+    """Id of the book"""
+
+    cfi: str
+    """Location within than book."""
```

Comment:
This helper class extracts some functionality that's used in multiple places and solves a dependency problem with the service and client as both need access to this.